libnmstate/appliers/linux_bridge.py

```diff
@@ -0,0 +1,41 @@
+#
+# Copyright 2018 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+
+import six
+
+
+def get_slaves_from_state(state, default=()):
+    ports = state.get('bridge', {}).get('port')
+    if ports is None:
+        return default
+    return [p['name'] for p in ports]
+
+
+def set_bridge_ports_metadata(master_state, slave_state):
+    _set_common_slaves_metadata(master_state, slave_state)
+
+    ports = master_state.get('bridge', {}).get('port', [])
+    port = next(
+        six.moves.filter(lambda n: n['name'] == slave_state['name'], ports),
+        {}
+    )
+    slave_state['_brport_options'] = port
+
+
+def _set_common_slaves_metadata(master_state, slave_state):
+    slave_state['_master'] = master_state['name']
+    slave_state['_master_type'] = master_state['type']
```

Comment:
- The `nm` package is indented for NetworkManager specific business logic, this code is API format processing, independent on any specific provider.
- The `netapplier` module needs to be split in a re-factoring effort, it is already too big, includes many unrelated things and therefore hard to read an maintain. It should end up as a glue of the general nmstate setup flow, with the idea that all individual steps will register into the flow.

For now I considered placing in the `appliers` package modules that handle specific focused subjects (for now, per iface type I guess, but it can evolve) that are not provider specific.
If you see a better split, you are welcome to suggest it.

Reply:
Instead of using a directory it could also just be a file like `libnmstate/linux_bridge_applier.py` - subdirectories with only single files are IMHO more confusing. But if there are going to be more files in there, soon, it is probably OK.

Reply:
I need to place there at least these for the metadata: `bond`, `ovs`, `team`.
Then, I want to place there anything that requires specific iface type logic but is not NM specific, not sure how much of this is there beyond the metadata stuff.
So yes, we will have there more soon.